contracts/FeeDisbursal/FeeDisbursal.sol

```diff
@@ -0,0 +1,269 @@
+    {   
+        // we don't do to a verified check here, this serves purely to check how much is claimable for an address
+        return claimablePrivate(token, spender, until);
+    }
+
+    /// @notice claim a token, to be called an investor
+    /// @param tokens addresses of the claimable token
+    /// @param spender address of the spender that would receive the funds
+    function claimableMutiple(address[] tokens, address spender)
+    public
+    view
+    returns (uint256[])
+    {   
+        // we don't to do a verified check here, this serves purely to check how much is claimable for an address
+        uint256[] memory result = new uint256[](tokens.length);
+        for (uint256 i = 0; i < tokens.length; i += 1) {
+            (uint256 claimableAmount, uint256 lastIndex) = claimablePrivate(tokens[i], spender, UINT256_MAX);
+            result[i] = claimableAmount;
+        }
+        return result;
+    }
+
+    /// @notice claim a token, to be called an investor
+    /// @param token address of the recyclable token
+    /// @param investors list of investors we want to recycle tokens for
+    /// @param until until what index to claim to
+    function recycle(address token, address[] investors, uint256 until)
+    public
+    {        
+        // @TODO: Recycle funds
+        // @TODO: add log message
+    }
+
+    // implementation of tokenfallback
+    function tokenFallback(address wallet, uint256 amount, bytes data)
+        public
+        only(ROLE_DISBURSER)
+    {
+        // @TODO: move access control to Controller
+        require(isDisbursableToken(msg.sender), "NF_DISB_UKNOWN_TOKEN");
+        require(amount > 0, "NF_DISB_ZERO_AMOUNT");
+
+        // cast and check pro rata token
+        BasicSnapshotToken proRataToken = BasicSnapshotToken(decodeAddress(data));
+        uint256 snapshotId = proRataToken.currentSnapshotId() - 1;
+        require(proRataToken.totalSupplyAt(snapshotId) > 0, "");
+
+        // create a new disbursal entry
+        _disbursals[msg.sender].push(Disbursal({
+            recycleAfterTimestamp: block.timestamp,
+            amount: amount,
+            proRataToken: proRataToken,
+            snapshotId: snapshotId,
+            disburser: wallet
+        }));
+        //@TODO: add log message
+    }
+
+    // implementation of reclaimbale
+    function reclaim(IBasicToken token)
+        public
+        only(ROLE_RECLAIMER)
+    {   
+        // forbid reclaiming any or our platform payment tokens
+        // @TODO: move access control to Controller
+        require(!isDisbursableToken(address(token)));
+        Reclaimable.reclaim(token);
+    }
+
+    /// @notice helper to determine if the token at the given address is supported for disbursing
+    /// @param token address of token in question
+    function isDisbursableToken(address token)
+        public
+        view
+        returns (bool)
+    {   
+        // @TODO: migrate this to new, more flexible token registering in universe
```

Comment:
move to controller
those could be payment tokens and asset tokens via respective collections
we'll also put NEU in the asset tokens